libindi/drivers/focuser/lacerta_mfoc.cpp

```diff
@@ -0,0 +1,649 @@
+/*******************************************************************************
+  Copyright(c) 2018 Franck Le Rhun. All rights reserved.
+  Copyright(c) 2018 Christian Liska. All rights reserved.
+ 
+ This library is free software; you can redistribute it and/or
+ modify it under the terms of the GNU Library General Public
+ License version 2 as published by the Free Software Foundation.
+ .
+ This library is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ Library General Public License for more details.
+ .
+ You should have received a copy of the GNU Library General Public License
+ along with this library; see the file COPYING.LIB.  If not, write to
+ the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ Boston, MA 02110-1301, USA.
+*******************************************************************************/
+
+#include "lacerta_mfoc.h"
+#include "indicom.h"
+#include "connectionplugins/connectionserial.h"
+
+#include <cmath>
+#include <memory>
+#include <cstring>
+#include <unistd.h>
+
+// We declare an auto pointer to lacerta_mfoc.
+std::unique_ptr<lacerta_mfoc> Lacerta_mfoc(new lacerta_mfoc());
+
+// The Focuser takes 5 ms to move for each step, completing 5000 steps in 25 seconds
+#define FOCUS_MOTION_DELAY 5
+// Delay for receiving messages
+#define FOCUSMFOC_TIMEOUT  1000
+// According to documentation for v2
+#define MFOC_POSMAX_HARDWARE 250000
+#define MFOC_POSMIN_HARDWARE 300
+
+void ISPoll(void *p);
+
+void ISGetProperties(const char *dev)
+{
+    Lacerta_mfoc->ISGetProperties(dev);
+}
+
+void ISNewSwitch(const char *dev, const char *name, ISState *states, char *names[], int n)
+{
+    Lacerta_mfoc->ISNewSwitch(dev, name, states, names, n);
+}
+
+void ISNewText(const char *dev, const char *name, char *texts[], char *names[], int n)
+{
+    Lacerta_mfoc->ISNewText(dev, name, texts, names, n);
+}
+
+void ISNewNumber(const char *dev, const char *name, double values[], char *names[], int n)
+{
+    Lacerta_mfoc->ISNewNumber(dev, name, values, names, n);
+}
+
+void ISNewBLOB(const char *dev, const char *name, int sizes[], int blobsizes[], char *blobs[], char *formats[],
+               char *names[], int n)
+{
+    INDI_UNUSED(dev);
+    INDI_UNUSED(name);
+    INDI_UNUSED(sizes);
+    INDI_UNUSED(blobsizes);
+    INDI_UNUSED(blobs);
+    INDI_UNUSED(formats);
+    INDI_UNUSED(names);
+    INDI_UNUSED(n);
+}
+
+void ISSnoopDevice(XMLEle *root)
+{
+    Lacerta_mfoc->ISSnoopDevice(root);
+}
+
+/************************************************************************************
+ *
+************************************************************************************/
+lacerta_mfoc::lacerta_mfoc()
+{
+  //  setSupportedConnections(CONNECTION_SERIAL);
+    FI::SetCapability(FOCUSER_CAN_ABS_MOVE | FOCUSER_CAN_REL_MOVE);
```

Comment:
Are you sure it cannot abort? How about Sync?

Reply:
Neither Abort nor Sync function according to Lacerta GmbH. I will suggest these as future functionalities.